WeatherService.Tests/InputTesting/JsonWeatherParserShould.cs

```diff
@@ -0,0 +1,45 @@
+﻿using FluentAssertions;
+using WeatherService.Input;
+using WeatherService.Utilities;
+
+namespace WeatherService.Tests;
+public class JsonWeatherParserTests
+{
+    private readonly JsonWeatherParser _sut;
+    public JsonWeatherParserTests()
+    {
+        _sut = new JsonWeatherParser();
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("")]
+    [InlineData("   ")]
+    public void ParseWeatherData_WhenInputIsNullOrEmpty_ShouldThrowArgumentNullException(string input)
+    {
+        _sut.Invoking(x => x.ParseWeatherData(input))
```

Comment:
It's generally more readable to follow the AAA (Arrange-Act-Assert) pattern, though it's not a strict obligation.